prediction_market_agent/agents/think_thoroughly_agent/think_thoroughly_agent.py

```diff
@@ -209,6 +227,21 @@ def generate_prediction_for_one_outcome(
             )
             return None
 
+    def get_correlated_markets(self, question: str) -> list[CorrelatedMarketInput]:
+        nearest_questions = self.pinecone_handler.find_nearest_questions(
+            5, text=question
```

Comment:
Did you play with this? Do you have some examples of what markets it fetches, please?

I'm thinking that if there are no similar markets, this way, it always returns 5 of them, even if they aren't similar at all. Maybe there should be some similarity threshold.

Reply:
Good point - that's true, not-so-correlated markets would be returned in this scenario.
I will add a score threshold here (already returned by Pinecone).

Reply:
Implemented a filter with default `threshold=0.7` (rule-of-thumb value from a post), I think that already helps

Reply:
Are we getting any relevant markets from Omen with this?

Reply:
We are fetching relevant markets - for ex, there are 2 markets related to Panama's election that can be cross-referenced.